clippy_lints/src/is_unit_expr.rs

```diff
@@ -132,6 +132,7 @@ fn is_unit_expr(expr: &Expr) -> Option<Span> {
 }
 
 fn check_last_stmt_in_block(block: &Block) -> bool {
+    if block.stmts.is_empty() { return false; }
```

Comment:
Can't you just return `true` and it'll report the `{}`? If the error message doesn't match, you can create an enum (Ignore, Semi, EmptyBlock) and return that here.